### ISSUE TITLE
Nested outputter not properly displaying unicode characters

### ISSUE THREAD

**Issue by terminalmage** [saltstack/salt#24344]
Assuming a directory containing the following files:

![pasted_image_at_2015_06_02_03_15_pm](https://cloud.githubusercontent.com/assets/328598/7947572/8c95551e-0943-11e5-9cb9-8575e4b90b56.png)

When running a `dir` on the directory, the unicode characters are not printed properly. The minion log contains the following information:

```
[INFO    ] Executing command 'dir c:\\unicode' in directory 'C:\\Users\\Shane'
[DEBUG   ] output:  Volume in drive C is OS
 Volume Serial Number is C079-E116

 Directory of c:\unicode

06/02/2015  03:08 PM    <DIR>          .
06/02/2015  03:08 PM    <DIR>          ..
06/02/2015  03:07 PM                 0 cyrllic_?.txt
06/02/2015  03:00 PM                 0 test∩┐╜.txt
06/02/2015  03:02 PM                 0 test∩┐╜.txt
06/02/2015  03:03 PM                 0 test∩┐╜.txt
               4 File(s)              0 bytes
               2 Dir(s)  248,701,345,792 bytes free
```

And the data printed to the console on the master is as follows:

```
root@shanelee-master:~# salt 'g*' cmd.shell 'dir c:\unicode'
gringo:
     Volume in drive C is OS
     Volume Serial Number is C079-E116

     Directory of c:\unicode

    06/02/2015  03:08 PM    <DIR>          .
    06/02/2015  03:08 PM    <DIR>          ..
    06/02/2015  03:07 PM                 0 cyrllic_?.txt
    06/02/2015  03:00 PM                 0 test.txt
    06/02/2015  03:02 PM                 0 test.txt
    06/02/2015  03:03 PM                 0 test.txt
                   4 File(s)              0 bytes
                   2 Dir(s)  248,701,345,792 bytes freeroot@shanelee-master:~# salt 'g*' cmd.shell 'dir c:\unicode'
```

This is happening with a minion running the latest from the 2015.5 branch.

**Comment by s0undt3ch** [comment]
This would be a good issue to attempt to use the `__salt_system_encoding__` global variable added. Can't remember if its available in 2015.5 though.

The idea is to decode the output using that encoding...

**Comment by twangboy** [comment]
We've been running into this problem lately in the windows side dealing with unicode characters. This is a codepage problem. The windows cmd prompt uses cp437 by default which is mostly latin characters with a few additional characters such as accents, diaeresis, etc. Python doesn't have a way to translate to or from cp437. You never know what you're going to get when you do a cmd.run with some unicode. It works the other way as well when python tries to read the stdout. The fix for some of our modules has been to transition from shelling out to using the windows API's as much as possible. That won't work for modules that depend on shelling out commands such as chocolatey, gem, etc.  I've seen some sites recommend changing the default code page for cmd to cp65001 and changing to a font that has all the characters for that code page such as `Lucida Console`, but I haven't tested that. Is there a way to do a `chcp 65001` and change the default font whenever we shell out?

**Comment by markuskramerIgitt** [comment]
Yes, @twangboy. 
I recently was surprised to find (in the tasklist) that ```cmd.run 'dir c:\unicode'``` becomes  ```chcp 437 & dir c:\unicode```

So I just tried  ```'chcp 65001 & dir c:\unicode'```
This should become ```chcp 437 & chcp 65001  & dir c:\unicode```

I ran these three commands (the last two are identical):
```
$ sudo salt MINION cmd.run 'chcp 65001 & chcp & dir \'
MINION:
    Aktive Codepage: 65001.
    Aktive Codepage: 65001.
     Datentr�ger in Laufwerk C: ist Windows


$ sudo salt MINION cmd.run 'chcp & dir \'
MINION:
    Aktive Codepage: 437.
     Datenträger in Laufwerk C: ist Windows


$ sudo salt MINION cmd.run 'chcp & dir \'
MINION:
    Aktive Codepage: 437.
     Datentr�ger in Laufwerk C: ist Windows

$
```
Please note that the last two commands are identical but the results are different.
From this little test I guess that ```chcp 65001 & ``` changes the copepage of the *next* cmd.run (which would be a bug in itself).
I further guess that codepage 65001 is a good idea.

I would recommend a modification where  Salt does not prepend ```chcp 437 &``` to ```cmd.run``` (What is the reason to use such an ancient codepage?).  
Codepage 65001 is UTF-8 and the default of Windows 7.

This bug has low priority to me because I can restrict my file names to ASCII.

**Comment by twangboy** [comment]
@markuskramerIgitt The results of your tests are interesting. The default code page may be dependent on your locale. My Windows 7 box is showing 437. So does my 8.1 and 10 box.

I think they went with 437 to try to standardize the return or something. It may be a good idea to default to 65001 instead to better handle the unicode. Now we'll need to figure out whether we want to do that in each command call or set a default in the registry or an autoexec.bat or something (if windows still handles that).

**Comment by terminalmage** [comment]
This was fixed in 2018.3.0
